src/posix/main.c

```diff
@@ -189,6 +198,9 @@ static void ParseArg(int aArgCount, char *aArgVector[], PosixConfig *aConfig)
     aConfig->mPlatformConfig.mSpeedUpFactor       = 1;
     aConfig->mLogLevel                            = OT_LOG_LEVEL_CRIT;
     aConfig->mPlatformConfig.mInterfaceName       = OPENTHREAD_POSIX_CONFIG_THREAD_NETIF_DEFAULT_NAME;
+#if OPENTHREAD_CONFIG_PLATFORM_NETIF_ENABLE
+    aConfig->mPlatformConfig.mTunDevice = NULL;
```

Comment:
Should we set `aConfig->mPlatformConfig.mTunDevice = OPENTHREAD_POSIX_TUN_DEVICE ` here to avoid checking whether the `aConfig->mPlatformConfig.mTunDevice == nullptr` before using the `aConfig->mPlatformConfig.mTunDevice`?